File: scripts/analyze_sales.py
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import os

# Create outputs directory if it doesn't exist
os.makedirs('../outputs', exist_ok=True)

# Read the data
df = pd.read_csv('../data/customer_sales.csv')

# Calculate summary statistics
summary_stats = df.describe()
gender_stats = df.groupby('gender')['total_spent'].agg(['mean', 'count'])

# Save summary statistics to CSV
summary_stats.to_csv('../outputs/summary_statistics.csv')
gender_stats.to_csv('../outputs/gender_statistics.csv')

# Create a correlation heatmap
plt.figure(figsize=(10, 8))
correlation_matrix = df[['age', 'income', 'items_purchased', 'total_spent']].corr()
sns.heatmap(correlation_matrix, annot=True, cmap='coolwarm', center=0)
plt.title('Correlation Heatmap of Customer Metrics')
plt.tight_layout()
plt.savefig('../outputs/correlation_heatmap.png')
plt.close()

# Create income vs total spent scatter plot
plt.figure(figsize=(10, 6))
sns.scatterplot(data=df, x='income', y='total_spent', hue='gender')
plt.title('Income vs Total Spent by Gender')
plt.xlabel('Income ($)')
plt.ylabel('Total Spent ($)')
plt.tight_layout()
plt.savefig('../outputs/income_vs_spent.png')
plt.close() 

File: scripts/analyze_sales.R
# Load required packages
library(tidyverse)
library(docopt)

remotes::install_github("Nicccolas/sales_package")
library(salespackage)

"This script loads and cleans the dataset.

Usage:
  analyze_sales.R --input=<input_file> --output_dir=<output_dir>

Options:
  --input=<input_file>    Path to raw data CSV file
  --output_dir=<output_dir>   Path to save outputs
" -> doc

opt <- docopt(doc)

analyze_sales <- function(input_file, output_dir) {
    # Read and analyze data
    df <- read_csv(input_file)

    # Basic summary statistics
    summary_stats <- summary(df)
    write_csv(as.data.frame(summary_stats), file.path(output_dir, "summary_statistics.csv"))

    # Simple scatter plot
    scatter_plot <- scatter(df, x = "income", y = "total_spent")
    ggsave(file.path(output_dir, "income_vs_spent.png"), scatter_plot, width = 8, height = 6) 

}

analyze_sales(opt$input, opt$output_dir)